### DISCUSSION TITLE
Most data not saved in URL

### DISCUSSION THREAD

**Discussion by maanemann** [SimeonGriggs/tailwind-css-palette-generator#63] [General]
First of all, i can't get over how cool this tool is. But it simply isn't possible to properly reopen a palette to adjust or view it, after you close or refresh the page. This is true for individual palette URLs as well as for collections.

The only data that's saved to the URL is the palette names and the manually entered base HEX value for each color. When you open the URL in a new session, all data, eg. HSL, is reset to default. And the colors change order (would be nice if you could change the order manually too).

If you look at the URL while adjusting one of the values that aren't carried over, you can see that it doesn't change. And it isn't updated when you press the 'copy URL' button either, as one might think. But if you change a manually entered base HEX value, or add or remove a color, it's reflected in the URL instantly.

Luckily all the individual shades are saved in the data object generated to paste into tailwind config, as well as in the API. But unfortunately you can't reverse the process and paste this back into the editor. Which would probably be very tricky to make possible too. So the only way to edit it later on is the manual way.

So as for now there's really no reason to copy an URL at all. I would contribute, if i had the skills, but can't say i do. Hope you can make this work eventually 🙏

**Comment by SimeonGriggs** [comment]
Not wrong, one of the many things I've wanted to add. Good nudge to get back to it. Thanks for the kind words!

**Reply by maanemann** [reply]
You're welcome, it's a wonderful tool! Hope you succeed with making the save feature work eventually :)

**Comment by maanemann** [comment]
If it has interest, i have added the data as comments in the file i keep my palettes in. An alternative to saving it in the URL could be to add this data automatically in the snippet. So it's easier to recreate. Also, if users could upload a file, maybe it's easier to extract the properties that way than the URL?

Just wanted to share those ideas, maybe it doesn't help, but i hope it does.

I have formated my palette file like so:


![image](https://github.com/SimeonGriggs/tints.dev/assets/15720449/f1953794-2b52-448c-ace8-0f12ed894b4a)

**Comment by realanthonyc** [comment]
Such a great tool! thanks a lot.
It would be great if light max and min values can be passed to URL as well.
